crates/orchestrator/src/tests/workers/proving/mod.rs

```diff
@@ -0,0 +1,134 @@
+use crate::config::config_force_init;
+use crate::database::MockDatabase;
+use crate::jobs::constants::JOB_METADATA_CAIRO_PIE_PATH_KEY;
+use crate::jobs::types::{ExternalId, JobItem, JobStatus, JobType};
+use crate::queue::MockQueueProvider;
+use crate::tests::common::init_config;
+use crate::workers::proving::ProvingWorker;
+use crate::workers::Worker;
+use da_client_interface::MockDaClient;
+use httpmock::MockServer;
+use mockall::predicate::eq;
+use prover_client_interface::MockProverClient;
+use rstest::rstest;
+use std::collections::HashMap;
+use std::error::Error;
+use std::time::Duration;
+use tokio::time::sleep;
+use uuid::Uuid;
+
+#[rstest]
+#[case(false)]
+#[case(true)]
+#[tokio::test]
+async fn test_proving_worker(#[case] incomplete_runs: bool) -> Result<(), Box<dyn Error>> {
+    let server = MockServer::start();
+    let da_client = MockDaClient::new();
+    let mut db = MockDatabase::new();
+    let mut queue = MockQueueProvider::new();
+    let mut prover_client = MockProverClient::new();
+
+    sleep(Duration::from_secs(10)).await;
+
+    const JOB_PROCESSING_QUEUE: &str = "madara_orchestrator_job_processing_queue";
+
+    // Mocking Prover Client
+
+    if incomplete_runs {
+        let jobs_vec_temp: Vec<JobItem> =
+            get_job_item_mock_by_id_vec(5).into_iter().filter(|val| val.internal_id != "3").collect();
+        // Mocking db call for getting successful snos jobs
+        db.expect_get_successful_snos_jobs_without_proving()
+            .times(1)
+            .with()
+            .returning(move || Ok(jobs_vec_temp.clone()));
+
+        let num_vec: Vec<i32> = vec![1, 2, 4, 5];
+
+        for i in num_vec {
+            db_checks(i, &mut db);
+        }
+
+        prover_client.expect_submit_task().times(4).returning(|_| Ok("task_id".to_string()));
+    } else {
+        for i in 1..5 + 1 {
+            db_checks(i, &mut db);
+        }
+
+        // Mocking db call for getting successful snos jobs
+        db.expect_get_successful_snos_jobs_without_proving()
+            .times(1)
+            .with()
+            .returning(move || Ok(get_job_item_mock_by_id_vec(5)));
+
+        prover_client.expect_submit_task().times(5).returning(|_| Ok("task_id".to_string()));
+    }
+
+    // Queue function call simulations
+    queue
```

Comment:
should we make this times 5 or 4?

Reply:
in first case it is 5 in second case when one job is unsuccessful that's why we are using 4 here.